scipy/interpolate/_rbfinterp.py

```diff
@@ -0,0 +1,538 @@
+            (nmonos, degree, ndim)
+            )
+
+    return y, d, smoothing, kernel, epsilon, degree, k
+
+
+class RBFInterpolator:
+    """
+    Radial basis function (RBF) interpolation in N dimensions
+
+    Parameters
+    ----------
+    y : (P, N) array_like
+        Data point coordinates
+    d : (P, ...) array_like
+        Data values at `y`
+    smoothing : float or (P,) array_like, optional
+        Smoothing parameter. The interpolant perfectly fits the data when this
+        is set to 0. For large values, the interpolant approaches a least
+        squares fit of a polynomial with the specified degree.
+    kernel : str, optional
+        Type of RBF. This should be one of:
+
+            - 'linear'                       : ``-r``
+            - 'tps' (thin plate spline)      : ``r**2 * log(r)``
+            - 'cubic'                        : ``r**3``
+            - 'quintic'                      : ``-r**5``
+            - 'mq' (multiquadratic)          : ``-sqrt(1 + r**2)``
+            - 'imq' (inverse multiquadratic) : ``1/sqrt(1 + r**2)``
+            - 'iq' (inverse quadratic)       : ``1/(1 + r**2)``
+            - 'ga' (Gaussian)                : ``exp(-r**2)``
+
+    epsilon : float, optional
+        Shape parameter that scales the input to the RBF. This can be ignored
+        if `kernel` is 'linear', 'tps', 'cubic', or 'quintic' because it has
+        the same effect as scaling the smoothing parameter. This must be
+        specified if `kernel` is 'mq', 'imq', 'iq', or 'ga'.
+    degree : int, optional
+        Degree of the added polynomial. For some RBFs the interpolant may not
+        be well-posed if the polynomial degree is too small. Those RBFs and
+        their corresponding minimum degrees are:
+
+            - 'mq'      : 0
+            - 'linear'  : 0
+            - 'tps'     : 1
+            - 'cubic'   : 1
+            - 'quintic' : 2
+
+        The default value is the minimum degree for `kernel` or 0 if there is
+        no minimum degree. Set this to -1 for no added polynomial.
+
+    Notes
+    -----
+    An RBF is a scalar valued function in N-dimensional space whose value at
+    :math:`x` can be expressed in terms of :math:`r=||x - c||`, where :math:`c`
+    is the center of the RBF.
+
+    An RBF interpolant for the vector of observations :math:`d`, which are made
+    at the locations :math:`y`, is a linear combination of RBFs centered at
+    :math:`y` plus a polynomial with a specified degree. The RBF interpolant is
+    written as
+
+    .. math::
+        f(x) = K(x, y) a + P(x) b
+
+    where :math:`K(x, y)` is a matrix of RBFs with centers at :math:`y`
+    evaluated at the interpolation points :math:`x`, and :math:`P(x)` is a
+    matrix of monomials, which span polynomials with the specified degree,
+    evaluated at :math:`x`. The coefficients :math:`a` and :math:`b` are the
+    solution to the linear equations
+
+    .. math::
+        (K(y, y) + \\lambda I) a + P(y) b = d
```

Comment:
```suggestion
        (K(y, y) + \\lambda I) a + P(y) b = d,
```

Reply:
I agree with the above suggested comma, but I think this one is good as is. You would not write "a and b are the solution to the linear equations X, and Y, where ..."